environment_setup/setup_software.sh

```diff
@@ -195,15 +195,25 @@ sudo apt-get install -y software-properties-common # required for add-apt-reposi
 sudo add-apt-repository ppa:ubuntu-toolchain-r/test -y
 # Required to make sure we install protobuf version 3.0.0 or greater
 sudo add-apt-repository ppa:maarten-fonville/protobuf -y
+
+# Running a PPA setup script to give us access to the correct node and yarn version
+# on non-Ubuntu 18.04 systems. 
+curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+
 sudo apt-get update
 
 host_software_packages=(
     g++-7 # We need g++ 7 or greater to support the C++17 standard
     python-rosinstall
     protobuf-compiler
     libprotobuf-dev
+    nodejs # Installed directly instead of using rosdep due to the lack of a default PPA
+    yarn # Installed directly instead of using rosdep due to the lack of a default PPA 
 )
 sudo apt-get install ${host_software_packages[@]} -y
+sudo npm install -g yarn
```

Comment:
> Can we remove this since it's already in `host_software_packages` above?
> 
> Suggested change Beta 
> [Give us feedback](https://www.research.net/r/SuggestedChanges)
> sudo npm install -g yarn

This line was removed